### PR TITLE
feat: lookup nft passport number (#2)

### PR DESCRIPTION
**Pull request by aahna-ashina**
#2

Screen Shot 2022-06-17 at 6 38 43 PM

### CODE REVIEW

File: server/pages/api/downloadPass.ts [hunk 1]
```diff
@@ -52,27 +53,41 @@ export default function handler(req: NextApiRequest, res: NextApiResponse) {
     // TODO
 
     // Check that the address has a passport NFT
-    // TODO
+    const PassportIssuerContract = new web3.eth.Contract(PassportIssuer.abi, '0x279c0b6bfCBBA977eaF4ad1B2FFe3C208aa068aC')
```
Comment (johnmark13) [inline server/pages/api/downloadPass.ts:56]: The address should probably come from configuration

File: server/pages/api/downloadPass.ts [hunk 2]
```diff
@@ -52,27 +53,41 @@ export default function handler(req: NextApiRequest, res: NextApiResponse) {
...
+    const passportIdPromise = PassportIssuerContract.methods.passportId(address).call()
+    passportIdPromise.catch((error: any) => {
+        console.error('catch')
+        console.error(error)
+        res.status(400).json({
```
Comment (johnmark13) [inline server/pages/api/downloadPass.ts:61]: Combine the above two lines into a single log message.

File: server/pages/api/downloadPass.ts [hunk 3]
```diff
@@ -52,27 +53,41 @@ export default function handler(req: NextApiRequest, res: NextApiResponse) {
...
+    passportIdPromise.catch((error: any) => {
+        console.error('catch')
+        console.error(error)
+        res.status(400).json({ 
+            error: 'Passport ID not found for address'
+        })
+        return
+    })
+    passportIdPromise.then((result: any) => {
+        console.log('then')
```
Comment (johnmark13) [inline server/pages/api/downloadPass.ts:67]: I love logging everything for tracing, but do you want to add more detail to this log message?

**Review by johnmark13 — COMMENTED** [review]
This is really cool. As a separate task/PR I would extract everything from lines 16 through 47 (verifiying the wallet request) to the middleware (pre-auth before it gets here) or utility code so it can be used everywhere, be separately tested and reduce the responsibility of this class. But that is out of scope for this PR!

File: server/pages/api/downloadPass.ts (hunk 1 above)
Comment (aahna-ashina) [reply server/pages/api/downloadPass.ts]: > The address should probably come from configuration

@johnmark13 You means as a Vercel environment variable?

File: server/pages/api/downloadPass.ts (hunk 2 above)
Comment (aahna-ashina) [reply server/pages/api/downloadPass.ts]: @johnmark13 Agree with you.  Refactored the error handling.  Also put the whole method within a `try-catch` to make it more robust.

File: server/pages/api/downloadPass.ts (hunk 3 above)
Comment (aahna-ashina) [reply server/pages/api/downloadPass.ts]: > I love logging everything for tracing, but do you want to add more detail to this log message?

@johnmark13 Yes, changed to `console.log('then result:', result)`

**Comment by aahna-ashina** [comment]
> This is really cool. As a separate task/PR I would extract everything from lines 16 through 47 (verifiying the wallet request) to the middleware (pre-auth before it gets here) or utility code so it can be used everywhere, be separately tested and reduce the responsibility of this class. But that is out of scope for this PR!

@johnmark13 Thanks for the suggestion.  I added this as a refactoring task here: #25

File: server/pages/api/downloadPass.ts (hunk 1 above)
Comment (aahna-ashina) [reply server/pages/api/downloadPass.ts]: @johnmark13 Made an issue for it: #26

**Review by johnmark13 — COMMENTED** [review]
Already merged, wanted to leave an official, whoop of approval. Git mobile interface is grim, not sure if this approves the PR post merge or is just a comment.

**Pull request merged**